orderer/consensus/follower/follower_chain.go

```diff
@@ -51,16 +137,121 @@ func (*Chain) Errored() <-chan struct{} {
 }
 
 func (c *Chain) Start() {
-	//TODO skeleton
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.started || c.stopped {
+		c.logger.Debug("Not starting because: started=%v, stopped=%v", c.started, c.stopped)
+		return
+	}
+
+	c.started = true
+
+	go c.run()
+
+	c.logger.Info("Started")
 }
 
 func (c *Chain) Halt() {
-	//TODO skeleton
+	c.halt()
+
+	select {
+	case <-c.doneChan:
+	}
+}
+
+func (c *Chain) halt() {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.stopped {
+		c.logger.Debug("Already stopped")
+		return
+	}
+	c.stopped = true
+	close(c.stopChan)
+	c.logger.Info("Stopped")
 }
 
-// StatusReport returns the ClusterRelation & Status
+// StatusReport returns the ClusterRelation & Status.
 func (c *Chain) StatusReport() (types.ClusterRelation, types.Status) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	clusterRelation := types.ClusterRelationFollower
+	if c.joinBlock != nil {
+		if err := c.amIInChannel(c.joinBlock); err == nil {
+			clusterRelation = types.ClusterRelationMember
+		}
+	}
+
 	status := types.StatusActive
-	//TODO if (height is >= join-block.height) return "active"; else return "onboarding"
-	return types.ClusterRelationFollower, status
+	if (c.joinBlock != nil) && (c.joinBlock.Header.Number >= c.support.Height()) {
+		status = types.StatusOnBoarding
+	}
+
+	return clusterRelation, status
+}
+
+func (c *Chain) run() {
+	c.logger.Debug("Enter")
```

Comment:
not a big fan of this log pattern, especially when this is intended to be a long-running func... it would be pretty hard to pair the `Exit` with its `Enter`, no?

Reply:
Additionally, imagine what happens when the calling function is removed from the record. What's the context? What does it mean?

Reply:
I am open to suggestions... the intent here is not really to pair the `Enter` and `Exit`, but to be able to detect that the go-routine actually started after say, a Join event; or conversely, actually exited after on-boarding has completed, for example.
These maybe far separated in time so normally there is no need to pair them, but one would like to know that the go-routine had started or ended in correlation to other meaningful events.
 
@sykesm do you mean you prefer a longer message with the full context? like `follower.Chain puller go-routine is starting`?

Reply:
Done